pandas/tests/frame/methods/test_combine_first.py

```diff
@@ -353,3 +355,19 @@ def test_combine_first_with_asymmetric_other(self, val):
         exp = pd.DataFrame({"isBool": [True], "isNum": [val]})
 
         tm.assert_frame_equal(res, exp)
+
+
+@pytest.mark.parametrize("val", [pd.NaT, np.nan, None])
+def test_combine_first_timestamp_bug(val):
```

Comment:
can you parametrize this with Period & Timedelta as well (as likey they suffered the same issue)